### PR TITLE
Calculate efficiency for executed observations

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR adds ability to calculate survey efficiency for executed observations.

### CODE REVIEW

File: static/js/components/GcnSelectionForm.jsx [hunk 1]
```diff
@@ -277,7 +287,15 @@ const GcnSelectionForm = ({ gcnEvent, setSelectedLocalizationName }) => {
...
 
-  const url = createUrl(selectedInstrumentId, formDataState);
+  const gcnUrl = createGcnUrl(selectedInstrumentId, formDataState);
+  const simSurveyUrl = createSimSurveyUrl(
+    selectedInstrumentId,
+    formDataState,
+    locLookUp[selectedLocalizationId]
+  );
+
+  console.log("gcnEventObservations", gcnEventObservations);
```
Comment (Theodlz) [inline static/js/components/GcnSelectionForm.jsx:297]: remove console.log

File: static/js/ducks/observations.js [hunk 2]
```diff
@@ -81,6 +81,7 @@ messageHandler.add((actionType, payload, dispatch) => {
 
 export function fetchGcnEventObservations(dateobs, filterParams = {}) {
   filterParams.localizationDateobs = dateobs;
+  filterParams.numPerPage = 1000;
```
Comment (Theodlz) [inline static/js/ducks/observations.js:84]: Is there a reason for that particular number, or is the goal just to get a lot / all of them and 1000 should be enough ?

File: skyportal/handlers/api/observation.py [hunk 3]
```diff
@@ -1486,3 +1493,216 @@ def delete(self, instrument_id):
...
+        if not set(unique_filters).issubset(set(instrument.sensitivity_data.keys())):
+            return self.error('Need sensitivity_data for all filters present')
+
+        for filt in unique_filters:
+            if not {'exposure_time', 'limiting_magnitude', 'zeropoint'}.issubset(
+                set(list(instrument.sensitivity_data[filt].keys()))
+            ):
+                return self.error(
+                    f'Sensitivity_data dictionary missing keys for filter {filt}'
+                )
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/observation.py:1683]: yes good idea.

File: skyportal/handlers/api/observation.py [hunk 4]
```diff
@@ -1486,3 +1493,216 @@ def delete(self, instrument_id):
...
+            "injectionFilename",
+            'data/nsns_nph1.0e+06_mejdyn0.020_mejwind0.130_phi30.txt',
+        )
+
+        localization = (
+            Localization.query_records_accessible_by(
+                self.current_user,
+            )
+            .filter(Localization.dateobs == localization_dateobs)
+            .filter(Localization.localization_name == localization_name)
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/observation.py:1665]: yeah fixing this too.

File: static/js/components/GcnSelectionForm.jsx (hunk 1 above)
Comment (mcoughlin) [reply static/js/components/GcnSelectionForm.jsx:297]: oops.

File: static/js/ducks/observations.js (hunk 2 above)
Comment (mcoughlin) [reply static/js/ducks/observations.js:84]: yes exactly. At some point we should maybe rethink a bit the map display, where we could consider aggregating statistics across field to then display in tooltips.

**Comment by Theodlz** [comment]
Hi Michael. When switching from the gcn in 2018 to the one in 2019 i got this error as the page crashed: `Uncaught TypeError: Cannot read properties of undefined (reading 'dateobs')`.
Also, when clicking on the simsurvey button that is on the GcnSelectionForm the downloads fails saying "failed-server problem"

**Comment by Theodlz** [comment]
Some stuff is a bit bulky, I can't tell if it comes from this PR or another. Sometimes the app fails to submit an observation plan, but it is still displayed on the list, and when deleting it i get a backend error:

```
ERROR:tornado.application:Uncaught exception DELETE /api/observation_plan/3 (127.0.0.1)
HTTPServerRequest(protocol='http', host='localhost:5000', method='DELETE', uri='/api/observation_plan/3', version='HTTP/1.0', remote_ip='127.0.0.1')
Traceback (most recent call last):
  File "/home/theodlz/Templates/skyportal/env/lib/python3.9/site-packages/tornado/web.py", line 1702, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "/home/theodlz/alright/reviews/skyportal/baselayer/app/access.py", line 44, in wrapper
    return tornado.web.authenticated(method)(self, *args, **kwargs)
  File "/home/theodlz/Templates/skyportal/env/lib/python3.9/site-packages/tornado/web.py", line 3173, in wrapper
    return method(self, *args, **kwargs)
  File "/home/theodlz/alright/reviews/skyportal/skyportal/handlers/api/observation_plan.py", line 308, in delete
    api.delete(observation_plan_request.id)
  File "/home/theodlz/alright/reviews/skyportal/skyportal/facility_apis/observation_plan.py", line 389, in delete
    observation_plan = req.observation_plans[0]
IndexError: list index out of range
```

**Comment by mcoughlin** [comment]
@Theodlz Thanks for your comments. I think I cleaned up the plan deletion and the event switching. If you want a functional set of observations, I added demo data to the PR you can load on the observation page for GW190425 and can click the button.

**Pull request merged**